### DISCUSSION TITLE
SSH deploy w/ semantic-release in a GitHub Action

### DISCUSSION THREAD

**Discussion by schie** [semantic-release/semantic-release#1859] [Configuration] [answered]
Hello!

Does anyone have any guidance on how to use SSH deploy keys w/ semantic-release in a GitHub Action?

Per [the docs](https://github.com/semantic-release/semantic-release/blob/7581aa85843b5039d27b36d6f63e743b8ceaa82a/docs/recipes/github-actions.md#pushing-packagejson-changes-to-a-master-branch), there are a number of vulnerabilities w/ using a one's personal access token as the `GH_TOKEN` to push to protected branches.

There is [documentation for the Circle CI or Travis CI](https://github.com/semantic-release/semantic-release/blob/7581aa85843b5039d27b36d6f63e743b8ceaa82a/docs/recipes/git-auth-ssh-keys.md) regarding how to do this, but nothing for Actions.

The only way I can think to get the action to write to the protected branch is to have a machine user as part of the org w/ the necessary permissions, but that seems like a bit overkill.

thanks!

**Comment by gr2m** [comment, marked as answer]
For what it's worth, I use bot users all the time for things like this.

**Reply by schie** [reply]
yeah, I'll probably follow suit.

**Reply by schie** [reply]
@gr2m how are bot's permissions and protected branch set up?

I keep running into protected branch issues.
